gradio/components/dialogue.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from gradio.components.base import server
+from gradio.components.textbox import Textbox
+from gradio.data_classes import GradioModel, GradioRootModel
+from gradio.events import Events
+
+
+class DialogueLine(GradioModel):
+    speaker: str
+    text: str
+
+
+class DialogueModel(GradioRootModel):
+    root: list[DialogueLine] | str
+
+
+class Dialogue(Textbox):
+    """
+    Creates a dialogue components for users to enter dialogue between speakers.
+
+    Demos: dia_dialogue_demo
+    """
+
+    EVENTS = [
+        Events.change,
+        Events.input,
+        Events.submit,
+    ]
+
+    data_model = DialogueModel
+
+    def __init__(
+        self,
+        value: list[dict[str, str]] | Callable | None = None,
+        *,
+        speakers: list[str] | None = None,
+        formatter: Callable | None = None,
+        emotions: list[str] | None = None,
+        separator: str = " ",
+        label: str | None = "Dialogue",
+        info: str
+        | None = "Type colon (:) in the dialogue line to see the available emotion and intonation tags",
+        placeholder: str | None = "Enter dialogue here...",
```

Comment:
Do we really want to have default values for `info` and `placeholder`? Might not be suitable in the general case, so I think its better to let deveopers specify